pkg/virt-controller/services/template.go

```diff
@@ -615,6 +615,23 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 	resources.Requests = make(k8sv1.ResourceList)
 	resources.Limits = make(k8sv1.ResourceList)
 
+	// Set Default CPUs request
+	if !vmi.IsCPUDedicated() {
+		vcpus := int64(1)
+		if vmi.Spec.Domain.CPU != nil {
+			vcpus = hardware.GetNumberOfVCPUs(vmi.Spec.Domain.CPU)
+		}
+		cpuAllocationRatio := t.clusterConfig.GetCPUAllocationRatio()
+		if vcpus != 0 && cpuAllocationRatio > 0 {
+			val := float64(vcpus) / cpuAllocationRatio
```

Comment:
this isn't the same as the openstack logic you referenced. ```"cpu_allocation_ratio": cpu_allocation_ratio = guest_vCPUs / cores```

based on the other comments, i'd expect something like

```
val := cpuAllocationRatio * float64(vcpus)
```

So if i set ratio to 1, i'd get a 1 host core for each guest core.  If i set ratio to .5 i'd get 1 host core for ever 2 guest cores.

Reply:
In openstack too this is being represented as a number > 0 
https://docs.openstack.org/ocata/config-reference/compute/config-options.html (defaulted to 16.0)

Otherwise for a ratio of 1:16 the value will have to 0.0625 which is not helpful.

This is a ratio of the number of virtual CPUs to 1 physical.